### PR TITLE
11.0 mig mass editing

### PR DESCRIPTION
**Pull request by jnegrete**

### CODE REVIEW

File: mass_editing/ChangeLog.txt
Comment (yajo) [inline mass_editing/ChangeLog.txt]: I vote to delete this file. Hard to maintain, and we have git for this.

File: mass_editing/README.rst
Comment (yajo) [inline mass_editing/README.rst]: 11.0

File: mass_editing/README.rst
Comment (yajo) [inline mass_editing/README.rst]: server-ux

File: mass_editing/README.rst
Comment (yajo) [inline mass_editing/README.rst]: update template

File: mass_editing/README.rst
Comment (yajo) [inline mass_editing/README.rst]: https

File: mass_editing/README.rst
Comment (yajo) [inline mass_editing/README.rst]: https

File: mass_editing/README.rst
Comment (yajo) [inline mass_editing/README.rst]: remove

File: mass_editing/__manifest__.py
Comment (yajo) [inline mass_editing/__manifest__.py]: AGPL-3

File: mass_editing/i18n/mass_editing.pot
Comment (yajo) [inline mass_editing/i18n/mass_editing.pot]: delete this file

File: mass_editing/models/mass_object.py
Comment (yajo) [inline mass_editing/models/mass_object.py]: Remove `_()`

File: mass_editing/models/mass_object.py
Comment (yajo) [inline mass_editing/models/mass_object.py]: This should be achieved by a `ir.rule` instead.

File: mass_editing/README.rst
Comment (yajo) [inline mass_editing/README.rst]: 🔥

File: mass_editing/README.rst
Comment (yajo) [inline mass_editing/README.rst]: 🔥

File: mass_editing/README.rst
Comment (yajo) [inline mass_editing/README.rst]: AGPL

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by yajo** [comment]
Please could you also check that this migration is not affected by https://github.com/OCA/server-tools/issues/959?

**Comment by RoelAdriaans** [comment]
```
Fout:
TypeError: val is null

http://localhost/web/static/src/js/views/basic/basic_model.js:3325
Traceback:
_parseServerData/<@http://localhost/web/static/src/js/views/basic/basic_model.js:3325:29
_.forEach@http://localhost/web/static/lib/underscore/underscore.js:145:9
_parseServerData@http://localhost/web/static/src/js/views/basic/basic_model.js:3311:9
_fetchRecord/<@http://localhost/web/static/src/js/views/basic/basic_model.js:1838:17
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3276:86
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3283:73
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
Deferred/</deferred[tuple[0]]@http://localhost/web/static/lib/jquery/jquery.js:3321:33
rpc/</<@http://localhost/web/static/src/js/services/ajax_service.js:13:21
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3276:86
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
Deferred/</deferred[tuple[0]]@http://localhost/web/static/lib/jquery/jquery.js:3321:33
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3283:73
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3283:73
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
done@http://localhost/web/static/lib/jquery/jquery.js:9275:33
callback@http://localhost/web/static/lib/jquery/jquery.js:9685:57
```

Same error as @CasVissers has in https://github.com/OCA/server-tools/pull/1048#issuecomment-344580895
This dump is with debug=assets .

**Comment by jnegrete** [comment]
Could you explain me how can I reproduce this error?

**Comment by RoelAdriaans** [comment]
@jnegrete Just by using the module.. 
It seems to work for simple fields, but not for many2one, like the country.
See this video: https://youtu.be/I_2aAcgzyf8

**Comment by astirpe** [comment]
I got this while searching a field:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/opt/odoo/odoo-11.0/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo-11.0/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/odoo-server/addons/web/controllers/main.py", line 866, in search_read
    return self.do_search_read(model, fields, offset, limit, domain, sort)
  File "/opt/odoo/odoo-server/addons/web/controllers/main.py", line 888, in do_search_read
    offset=offset or 0, limit=limit or False, order=sort or False)
  File "/opt/odoo/odoo-11.0/odoo/models.py", line 4118, in search_read
    records = self.search(domain or [], offset=offset, limit=limit, order=order)
  File "/opt/odoo/custom/addons/mass_editing/models/ir_model_fields.py", line 18, in search
    list(map(int, domain[2][1:-1].split(','))))]
ValueError: invalid literal for int() with base 10: 'nvo'
```

**Comment by jnegrete** [comment]
@Yajo could you tell me if I forgot something about the requested changes?
I'm ready to squash my commits, what do you think?

**Review by yajo — APPROVED** [review]
I think you can proceed with the squash.

All my remarks were solved, although they were not much more than stylistic... I think somebody more experienced in this addon should review too, because I see some logical changes that I don't understand...

**Comment by astirpe** [comment]
The error I got seems not reproducible anymore after the last commit. But I didn't test the module yet.

**Comment by ddufresne** [comment]
:+1: functional review

**Comment by pedrobaeza** [comment]
Please squash all adjacent translation commits together, and you can also merge some commits by author/logic group. After that, I will merge. Thanks for the work!

File: mass_editing/i18n/am.po
Comment (JayVora-SerpentCS) [inline mass_editing/i18n/am.po:9]: Still has the old translation?

Comment (pedrobaeza) [reply mass_editing/i18n/am.po:9]: I think it's because headers are not updated unless you export from Odoo.

**Comment by pedrobaeza** [comment]
I have made myself the squash, so I finally merge this.

**Comment by rafaelbn** [comment]
There are 3 PRs related with this module opened in previous version, maybe same issue are in v11 if they are not taken in care

- https://github.com/OCA/server-tools/pull/848
- https://github.com/OCA/server-tools/pull/849 
- https://github.com/OCA/server-tools/pull/786

Just to point them

Regards

**Pull request merged**
